jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/WarFiles.java

```diff
@@ -17,32 +17,86 @@
    *     provided doesn't exist
    * @throws InvalidImageReferenceException if the base image reference is invalid
    */
-  public static JibContainerBuilder toJibContainerBuilder(ArtifactProcessor processor)
+  public static JibContainerBuilder toJibContainerBuilder(
+      ArtifactProcessor processor,
+      CommonCliOptions commonCliOptions,
+      CommonContainerConfigCliOptions commonContainerConfigCliOptions,
+      ConsoleLogger logger)
       throws IOException, InvalidImageReferenceException {
-    JibContainerBuilder containerBuilder = Jib.from("jetty");
+    JibContainerBuilder containerBuilder;
+    List<FileEntriesLayer> layers;
+    Optional<String> baseImage = commonContainerConfigCliOptions.getFrom();
+    if (baseImage.isPresent()) {
+      containerBuilder =
+          ContainerBuilders.create(
+              baseImage.get(), Collections.emptySet(), commonCliOptions, logger);
+    } else {
+      containerBuilder =
+          ContainerBuilders.create("jetty", Collections.emptySet(), commonCliOptions, logger);
+    }
 
-    // JVM Flags are ignored
-    List<String> entrypoint = processor.computeEntrypoint(ImmutableList.of());
+    List<String> programArguments =
+        commonContainerConfigCliOptions.getProgramArguments().isEmpty()
+            ? null
+            : commonContainerConfigCliOptions.getProgramArguments();
+    layers = processor.createLayers();
+    containerBuilder
+        .setEntrypoint(computeEntrypoint(commonContainerConfigCliOptions, processor))
+        .setProgramArguments(programArguments)
+        .setFileEntriesLayers(layers)
+        .setExposedPorts(commonContainerConfigCliOptions.getExposedPorts())
+        .setVolumes(commonContainerConfigCliOptions.getVolumes())
+        .setEnvironment(commonContainerConfigCliOptions.getEnvironment())
+        .setLabels(commonContainerConfigCliOptions.getLabels());
+    commonContainerConfigCliOptions.getUser().ifPresent(containerBuilder::setUser);
+    commonContainerConfigCliOptions.getFormat().ifPresent(containerBuilder::setFormat);
+    commonContainerConfigCliOptions.getCreationTime().ifPresent(containerBuilder::setCreationTime);
 
-    containerBuilder.setEntrypoint(entrypoint).setFileEntriesLayers(processor.createLayers());
     return containerBuilder;
   }
+
+  @Nullable
+  private static List<String> computeEntrypoint(
+      CommonContainerConfigCliOptions commonContainerConfigCliOptions, ArtifactProcessor processor)
+      throws IOException {
+    List<String> entrypoint = commonContainerConfigCliOptions.getEntrypoint();
+    if (!entrypoint.isEmpty()) {
+      return entrypoint;
+    }
+    Optional<String> baseImage = commonContainerConfigCliOptions.getFrom();
+    Boolean isDefaultBaseImage =
+        !baseImage.isPresent() || (baseImage.isPresent() && baseImage.get().startsWith("jetty"));
+    if (isDefaultBaseImage) {
+      return processor.computeEntrypoint(ImmutableList.of());
```

Comment:
Hmm... I cannot say I like this. I may not remember all the context exactly, but I feel the entire logic of this function should reside inside `processor.computeEntrypoint()`. The `processor` instance should be created with (or given at some point potentially including now) all the necessary information to be able to do its job to compute and give us an entyrpoint. Your thoughts?

Reply:
Ah I agree that it is not ideal to compute entrypoint in two places. It is definitely possible to have this logic in computeEntrypoint in the `processor` but I'm worried that this will blur the line between `WarFiles` and the processors. I would think the processor was more for opening up the artifact and creating layers whereas `WarFiles` was for accepting cli arguments. But I'm open to the option of adding this logic to `processor` if it is making the code confusing to read.

Reply:
Ah, I see. That makes sense. In that perspective, how about not calling `processor.computeEntrypoint()` at all? I see what `processor.computeEntrypoint()` doesn't have anything to do with processing the contents of a WAR. Also, it doesn't make sense that `processor.computeEntrypoint()` blindly returns an entrypoint that only works for `jetty`.

Reply:
That is a good point. `computeEntrypoint()` did look into the contents of a JAR but it doesn't serve the same purpose for WAR files. Now that I think about it, it may be better to have a separate `JarProcessor` and a `WarProcessor` instead of `ArtifactProcessor` since the only method that is implemented in the WAR case is `createLayers`? This would improve the readability by getting rid of the unused methods from `StandardWarExplodedProcessor`